go/dpagg/mean_test.go

```diff
@@ -738,3 +738,109 @@ func TestBMFloat64Serialization(t *testing.T) {
+			sumConfInt:   noise.ConfidenceInterval{LowerBound: 70.0, UpperBound: 100.0},
+			countConfInt: noise.ConfidenceInterval{LowerBound: 2.0, UpperBound: 5.0},
+			want:         noise.ConfidenceInterval{LowerBound: 9.0, UpperBound: 9.0},
+		},
+	} {
+		mean := NewBoundedMeanFloat64(tc.meanOpt)
+		mean.normalizedSum.noise = getMockConfInt(tc.sumConfInt)
+		mean.count.noise = getMockConfInt(tc.countConfInt)
+		mean.Result()
+		got, _ := mean.computeConfidenceIntervalForExplicitAlphaSplit(0.1, 0.05) // Parameters are ignored.
+		if !ApproxEqual(got.LowerBound, tc.want.LowerBound) {
+			t.Errorf("TestMeanComputeConfidenceIntervalForExplicitAlphaSplit(ConfidenceInterval{%f, %f})=%0.10f, want %0.10f, LowerBounds are not equal",
+				tc.meanOpt.Lower, tc.meanOpt.Upper, got.LowerBound, tc.want.LowerBound)
+		}
+		if !ApproxEqual(got.UpperBound, tc.want.UpperBound) {
+			t.Errorf("TestMeanComputeConfidenceIntervalForExplicitAlphaSplit(ConfidenceInterval{%f, %f})=%0.10f, want %0.10f, UpperBounds are not equal",
+				tc.meanOpt.Lower, tc.meanOpt.Upper, got.UpperBound, tc.want.UpperBound)
+		}
+	}
+}
+
+/**
+ * This test was designed to be deterministic. It checks that bounding is done correctly.
+ */
+func TestMeanComputeConfidenceInterval_confidenceBoundsAlwaysInsideProvidedBoundaries(t *testing.T) {
+	lower := 0.0
+	upper := 1.0
+	meanOpt := &BoundedMeanFloat64Options{Epsilon: 1.0, Delta: 0.123, Lower: lower, Upper: upper, Noise: noise.Gaussian(), MaxContributionsPerPartition: 1}
+	mean := NewBoundedMeanFloat64(meanOpt)
+	mean.Add(-1.0)
+	mean.Add(0.0)
+	mean.Add(1.0)
+	mean.Add(2.0)
+	mean.Result()
+	// Generate a random meanAlpha and a random split.
+	meanAlpha := rand.Uniform()
+	countAlpha := rand.Uniform() * meanAlpha
+	meanConfInt, _ := mean.computeConfidenceIntervalForExplicitAlphaSplit(meanAlpha, countAlpha)
+	if meanConfInt.LowerBound < lower || meanConfInt.LowerBound > upper {
+		t.Errorf("TestMeanComputeConfidenceInterval_confidenceBoundsAlwaysInsideProvidedBoundaries")
+	}
+	if meanConfInt.UpperBound < lower || meanConfInt.UpperBound > upper {
+		t.Errorf("TestMeanComputeConfidenceInterval_confidenceBoundsAlwaysInsideProvidedBoundaries")
+	}
+}
+
+// ComputeConfidenceInterval checks that the parameters passed are the ones we expect for sum.
+func (mn mockBMNoise) ComputeConfidenceIntervalFloat64(noisedX float64, l0 int64, lInf, eps, del, alpha float64) (noise.ConfidenceInterval, error) {
+	return noise.ConfidenceInterval{}, nil
+}
+
+// ComputeConfidenceIntervalInt64 checks that the parameters passed are the ones we expect for count.
+func (mn mockBMNoise) ComputeConfidenceIntervalInt64(noisedX, l0, lInf int64, eps, del, alpha float64) (noise.ConfidenceInterval, error) {
+	return noise.ConfidenceInterval{}, nil
+}
+
+func TestMeanComputeConfidenceInterval_callsNoiseComputeConfidenceIntervalCorrectly(t *testing.T) {
+	bmf := getMockBMF(t)
+	bmf.Result()
+	bmf.computeConfidenceIntervalForExplicitAlphaSplit(alphaLevel, alphaLevel/2)
+}
```

Comment:
I am not sure if this tests callsNoiseComputeConfidenceIntervalCorrectly since functions on Lines 825 and 830 are empty.

Refer to how this was done in the same file for addNoise: https://github.com/google/differential-privacy/blob/b8aacd8369c4a3a7d3ed0872231530edfdeecde7/go/dpagg/mean_test.go#L237

Also here is an example of how it was done in Count: [Example](https://github.com/google/differential-privacy/blob/514d8c56e7c1226116e08168a8e5bd8270767663/go/dpagg/count_test.go#L478-L531)